src/cohort.js

```diff
@@ -0,0 +1,96 @@
+class Cohort {
+  constructor() {
+    this.cohortList = []
+  }
+
+  createCohort(cohortName) {
+    const newCohort = { name: cohortName, students: [], capacity: 24 }
+    const findCohort = this.cohortList.find(
+      (cohort) => cohort.name === cohortName
+    )
+
+    if (findCohort) {
+      throw new Error('This Cohort already exists, please choose another name!')
+    }
+    if (!cohortName) {
+      throw new Error('Please give a name to your class!')
+    } else {
+      this.cohortList.push(newCohort)
+    }
+
+    return this.cohortList
+  }
+
+  searchCohort(cohortName) {
+    const findCohort = this.cohortList.find(
+      (cohort) => cohort.name === cohortName
+    )
+
+    if (!findCohort) {
+      throw new Error('Cohort not found')
+    }
+
+    return findCohort
+  }
+
+  searchStudent(cohortName, studentID) {
+    const findCohort = this.searchCohort(cohortName)
+    const findStudent = findCohort.students.find(
+      (student) => student.studentID === studentID
+    )
+
+    if (!findStudent) throw new Error('Student not found')
+
+    return findStudent
+  }
+
+  addStudent(cohortName, studentToAdd) {
+    const findCohort = this.searchCohort(cohortName)
+    const studentExists = findCohort.students.find(
+      (student) => student.studentID === studentToAdd.studentID
+    )
+
+    if (studentExists) {
+      throw new Error('this student already exists, add another student!')
+    }
+
+    for (let i = 0; i < this.cohortList.length; i++) {
+      for (let j = 0; j < this.cohortList[i].students.length; j++) {
```

Comment:
Nested for loops are usually a sign of code that needs improvement. With large data sets, this would take a very long time to run.

If you had a class for a `Cohort`, you could have a `findStudent` method on that class